_posts/2019-11-13-variational-convnet/variational_convnet.Rmd

```diff
@@ -0,0 +1,619 @@
+---
+title: "Variational convnets with tfprobability"
+description: > 
+ In a Bayesian neural network, layer weights are distributions, not tensors.
+ Using tfprobability, the R wrapper to TensorFlow Probability, we can build regular Keras models that have probabilistic layers, and thus get uncertainty estimates "for free".
+ In this post, we show how to define, train and obtain predictions from a probabilistic convolutional neural network.
+ 
+author:
+  - name: Sigrid Keydana
+    affiliation: RStudio
+    affiliation_url: https://www.rstudio.com/
+slug: keydana2019variationalconv
+bibliography: bibliography.bib
+date: 11-13-2019
+categories:
+  - Probability and statistics
+  - Time Series
+  - Keras
+output:
+  distill::distill_article:
+    self_contained: false
+preview: images/bbb.png
+---
+
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE, eval = FALSE)
+```
+
+
+A bit more than a year ago, in his beautiful [guest post](https://blogs.rstudio.com/tensorflow/posts/2018-07-17-activity-detection/) Nick Strayer showed how to classify a set of everyday activities using smartphone-recorded gyroscope and accelerometer data. Accuracy was very good, but Nick went on to inspect classification results more closely. Were there activities more prone to misclassification than others? And how about those erroneous results: Did the network report them with equal, or less, confidence than those that were correct?
+
+Technically, what we refer to when talking of _confidence_ in that manner is the _score_ obtained for the "winning" class after softmax activation ^[see [Winner takes all: A look at activations and cost functions](https://blogs.rstudio.com/tensorflow/posts/2018-10-11-activations-intro/)]. If that winning score is 0.9, we might say "the network is sure that's a gentoo penguin"; if it's 0.2, we'd instead conclude "to the network, neither option seemed fitting, but cheetah looked best".
```

Comment:
```suggestion
Technically, what we speak of _confidence_ in that manner, we're referring to the _score_ obtained for the "winning" class after softmax activation ^[see [Winner takes all: A look at activations and cost functions](https://blogs.rstudio.com/tensorflow/posts/2018-10-11-activations-intro/)]. If that winning score is 0.9, we might say "the network is sure that's a gentoo penguin"; if it's 0.2, we'd instead conclude "to the network, neither option seemed fitting, but cheetah looked best".
```

Reply:
I played around a little with the wording in the first sentence above — feel free to take it or leave it as you like! (This is also one of those weird cases where "speak" sounds more natural to me than "talk," but I don't have a rationale).

Reply:
hey, thx! I think you mean "when we speak of", so I'll skip the autocommit and edit this manually